### ISSUE TITLE
Баг с дистанцией до цели

### ISSUE THREAD

**Issue by DmitriySolontsevoy** [OGSR/OGSR-Engine#387]
Если вращать камерой, стоя на месте, расстояние до цели на радаре будет меняться.
![ss_dima_01-04-22_16-20-18_(l01_escape)](https://user-images.githubusercontent.com/27158522/148073644-7864f402-e837-4f60-815a-41761756efa5.png)
![ss_dima_01-04-22_16-20-13_(l01_escape)](https://user-images.githubusercontent.com/27158522/148073670-f946285a-2d3e-4993-8b02-c4d91a91a73f.png)
В оригинале такого не наблюдал

**Comment by KRodinn** [comment]
Не подтверждаю
![ss_admin_01-06-22_17-31-00_(l03_agroprom)](https://user-images.githubusercontent.com/16670637/148398733-a281f0bc-6bf3-4cbe-a295-64f1c0b21d57.png)
![ss_admin_01-06-22_17-31-03_(l03_agroprom)](https://user-images.githubusercontent.com/16670637/148398745-8f72a5a4-f7af-4b96-adc2-c87243118186.png)

**Comment by DmitriySolontsevoy** [comment]
Специально проверял на чистой игре + последний релиз OGSR

**Comment by KRodinn** [comment]
кстати да, есть. Но не всегда и не везде. ЧуднО.

**Comment by xrModder** [comment]
Подсказка: проблема в функции `CUIMapWnd::SetTargetMap`

**Comment by KRodinn** [comment]
Сомнительно

**Comment by v2v3v4** [comment]
Там берётся позиция Device.vCameraPosition: xrGame\map_location.cpp

**Comment by DmitriySolontsevoy** [comment]
В методе ``UpdateSpotPointer`` сделал вместо:
```cpp
Fvector2 tt = map->ConvertLocalToReal(m_position_on_map);
Fvector ttt;
ttt.set		(tt.x, 0.0f, tt.y);
float dist_to_target = Level().CurrentEntity()->Position().distance_to(ttt);
map->SetPointerDistance	(dist_to_target);
```
Вот так:
```cpp
Fvector2 tt = sp->MapLocation()->Position();
Fvector ttt;
ttt.set(tt.x, 0.0f, tt.y);
float dist_to_target = Actor()->Position().distance_to(ttt);
map->SetPointerDistance	(dist_to_target);
```

Полечило

**Comment by joye-ramone** [comment]
выглядит логично. я перенесу в репу

**Comment by joye-ramone** [comment]
Нет, не работает нормально. Вообще я этот баг не могу повторить. Нужно больше информации

**Comment by DmitriySolontsevoy** [comment]
@joye-ramone да, заметил. Мой вариант работает правильно только тогда, когда я нахожусь на той же локации, на которой метка цели. Если быть на другой локации, дистанция приведёт к рандомной точке на карте (предположительно к координатам, которые занимает цель на той локации, на которой она находится).
А по поводу воспроизведения бага. Специально только что повторил на свежем релизе OGSR и чистой игре. Первое же задание на кордоне пойти к Петрухе - при вращении камеры дистанция меняется

**Comment by joye-ramone** [comment]
Ок проверим.

**Comment by joye-ramone** [comment]
да, так и есть
1. сейчас в движке если точка на той же карте где ГГ - баг есть, но если точка на другой локе - указываються координаты до ближайшего перехода и бага нету
2. с твоим фиксом наоборо расстояние до точки на одной локе начинает работать верно, а для другой локи нет

Можно просто в лоб совместить в целом, ну или найти причину бага. Посмотрю что можно сдлеть
